centreon/src/App/MonitoringConfiguration/Infrastructure/Dbal/DbalCommandRepository.php

```diff
@@ -198,15 +194,22 @@ public function update(Command $command): void
     private function createCommand(array $row): Command
     {
         $command = $this->transformer->transform($row);
-
-        if ($row['connector_id'] !== null) {
-            $connectorId = $row['connector_id'];
-            $connector = $this->connectorRepository->findById(new ConnectorId($connectorId));
-            if ($connector instanceof Connector) {
-                $command->addConnector($connector);
-            }
-        }
-
-        return $command;
+        $connector = $this->connectorRepository->findByCommand($command);
```

Comment:
I don't understand the need for this diff
And what if command->connector is null ? the connectorRepository->findByCommand() doesn't seem to handle this case

Reply:
Initial need is principally for lazyness. But as the lazyness has been removed for this part it could be revert.

findByCommand handle null return as we find a connector by a command_id (that is defined), whether we have a Connector to return or we juste return null